### DISCUSSION TITLE
Style reference by image

### DISCUSSION THREAD

**Discussion by Codeschreibaer** [Acly/krita-ai-diffusion#392] [Q&A]
As i understand, "image" is now "reference". And image refers to content, not to style, of a reference image. Am i right?
Is there a way to refence the style mainly or exclusively? (i.e. with special prompt or another mode?) Since krita is for painting etc it would be nice. Regards. 
(New to here. Sorry of missing part of etiquette.)

**Comment by Danamir** [comment]
The "reference" control layer is using a IPAdapter node with the designated layer as source. So it should also refer to the style of the image. I don't know if some IPAdapter models are more useful for style than others.

**Comment by Acly** [comment]
I'm not aware of any model which does this. Reference (IP-Adapter) will embed and influence both style & content, they're not seperable. What might help is to use multiple images with very different content, but matching style, to "dilute" the content part while keeping the style consistent.

That's basically how style LoRA work, except they have the benefit of additional text descriptions for the content (so it can be subtracted) and training on a larger image set. If you want best results that's probably the way to go.

**Reply by Danamir** [reply]
If we add the `start_at` part of the IPAdapter node, we should be able to leave the image composition to the initial model, and impact only the final rendering part. But I'm really not sure this would be good for style transfer. Maybe for some artistic styles, but surely not all.

**Comment by Codeschreibaer** [comment]
if i understnd the follwing correctly, my qustioncovers directly tehessence of NST:

The NST (Neural Style Transfer) technique optimizes the output so that it matches the content of the content image and the style of the reference image. To extract the statistics of both images, NST uses a convolutional network. This image styling offers endless possibilities for designing, coloring, and creating amazing paintings.

seen on https://www.capcut.com/de-de/resource/neural-style-transfer 




On 09.02.24 13:32, Acly ***@***.***> wrote:
> I'm not aware of any model which does this. Reference (IP-Adapter) will 
> embed and influence both style & content, they're not seperable. What 
> might help is to use multiple images with very different content, but 
> matching style, to "dilute" the content part while keeping the style 
> consistent.
> 
> That's basically how style LoRA work, except they have the benefit of 
> additional text descriptions for the content (so it can be subtracted) 
> and training on a larger image set. If you want best results that's 
> probably the way to go.
> 
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/Acly/krita-ai-diffusion/discussions/392#discussioncomment-8418433>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AQ7WDGOXTPX3HYAIIU6WZHDYSYJOJAVCNFSM6AAAAABC7BZIFSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DIMJYGQZTG>.
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>
> 
> 


-- 
Diese E-Mail wurde von Avast-Antivirussoftware auf Viren geprüft.
www.avast.com
